content/docs/user-guide/integrations/huggingface.md

```diff
@@ -0,0 +1,119 @@
+you must have a Hugging Face account with a registered SSH key, regardless of
+whether the repository is public or private. Using Git over HTTPS to access
+public repositories does not require a Hugging Face account. Please refer to the
+Hugging Face [documentation](https://huggingface.co/docs/hub/security-git-ssh)
+for more information.
+
+This limitation applies to both `dvc get` and `dvc import`.
+
+</admon>
+
+### Setup
+
+```cli
+$ pip install dvc
+```
+
+### Download data
+
+`dvc get` can be used to download data from Hugging Face Hub. Downloading both
+individual files and directories is supported.
+
+Download the `sd_xl_base_1.0` model from
+[Stable Diffusion XL 1.0](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0):
+
+```cli
+$ dvc get https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0 sd_xl_base_1.0.safetensors
+```
+
+### Import data
+
+`dvc import` can be used to import data from Hugging Face Hub into a <abbr>DVC
+project</abbr>. Importing both individual files and directories is supported.
+
+Import the `sd_xl_base_1.0` model from
+[Stable Diffusion XL 1.0](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0):
+
+```cli
+$ dvc import https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0 sd_xl_base_1.0.safetensors
+```
+
+## Hugging Face Datasets
+
+[Hugging Face Datasets](https://huggingface.co/docs/datasets/index) is a Python
+library accessing and sharing datasets. The DVC Python API provides
+[DVCFileSystem](/doc/api-reference/dvcfilesystem), an fsspec-compatible
+filesystem which can be used with Hugging Face Datasets to load data from a
+<abbr>DVC project</abbr>. Please refer to the Hugging Face
+[documentation](https://huggingface.co/docs/datasets/main/en/filesystems) for
+more detailed information on using fsspec filesystems with Hugging Face
+Datasets.
+
+### Setup
+
+```cli
+$ pip install datasets dvc
+```
+
+### Load DVC data
+
+When DVC is installed in the same Python environment as Hugging Face Datasets,
+DVCFileSystem will be used automatically when a `dvc://` filesystem URL is
+provided.
+
+<admon type="tip">
+
+Note that `dvc://` URLs contain the path to the file you wish to load, relative
+the root of the DVC project. `dvc://` URLs should not contain a Git repository
+URL.
+
+The Git repository URL and any additional DVCFileSystem parameters are provided
+separately, through the `storage_options` dictionary.
+
+</admon>
+
+Load metrics (`eval/metrics.json`) from the
+[example-get-started](https://github.com/iterative/example-get-started)
+repository in Hugging Face Datasets:
+
+```python
+>>> from datasets import load_dataset
+>>> load_dataset(
+...   "json",
+...   data_files="dvc://eval/metrics.json",
+...   storage_options={"url": "https://github.com/iterative/example-get-started.git"},
+... )
+DatasetDict({
+    train: Dataset({
+        features: ['avg_prec', 'roc_auc'],
+        num_rows: 1
+    })
+})
+```
```

Comment:
Is `"json"` an arbitrary name? Does it need to include `json` to be read as a json dataset?

What am I doing wrong in this example?

```python
>>> load_dataset("parquet", data_files="dvc://github-issues/dataset.parquet", storage_options={"url": "https://github.com/iterative/dataset-registry.git"})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/dave/micromamba/envs/huggingface/lib/python3.11/site-packages/datasets/load.py", line 2128, in load_dataset
    builder_instance = load_dataset_builder(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dave/micromamba/envs/huggingface/lib/python3.11/site-packages/datasets/load.py", line 1814, in load_dataset_builder
    dataset_module = dataset_module_factory(
                     ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dave/micromamba/envs/huggingface/lib/python3.11/site-packages/datasets/load.py", line 1429, in dataset_module_factory
    ).get_module()
      ^^^^^^^^^^^^
  File "/Users/dave/micromamba/envs/huggingface/lib/python3.11/site-packages/datasets/load.py", line 957, in get_module
    data_files = DataFilesDict.from_patterns(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dave/micromamba/envs/huggingface/lib/python3.11/site-packages/datasets/data_files.py", line 686, in from_patterns
    DataFilesList.from_patterns(
  File "/Users/dave/micromamba/envs/huggingface/lib/python3.11/site-packages/datasets/data_files.py", line 591, in from_patterns
    resolve_pattern(
  File "/Users/dave/micromamba/envs/huggingface/lib/python3.11/site-packages/datasets/data_files.py", line 380, in resolve_pattern
    raise FileNotFoundError(error_msg)
FileNotFoundError: Unable to find 'dvc://github-issues/dataset.parquet'
```

Reply:
When loading a dataset from an arbitrary file (for supported types) you have to specify the name of the loader you are trying to use, so `json` for json files, `parquet` for parquet files (normally the first arg is treated as the name of a public huggingface repository to search for) https://huggingface.co/docs/datasets/package_reference/loading_methods#from-files

Your example looks like it should work, it's probably a dvcfs issue. I'll take a look into it

Reply:
On further investigation I don't think this is a problem on the DVC end. It's either a huggingface issue or possibly a problem with how the parquet file in the data registry was generated.

(Should note that a parquet file is actually a directory)

Downloading the dataset locally, both of these cases fail:
```
load_dataset("parquet", data_files="github-issues/dataset.parquet")
load_dataset("parquet", data_dir="github-issues/")
```

The `data_files` case fails with the same filenotfounderror you get with dvcfs, and the `data_dir` case fails for me with a pyarrow deserialization error (maybe the version installed with datasets is incompatible the version used to generate the dataset-registry parquet dir?)

Reply:
The local case still fails for me even using the explicit `data_files={"dataset": "dataset.parquet"}` format from the hf docs: https://huggingface.co/docs/datasets/main/en/loading#parquet

Reply:
Odd. Reading that dataset with https://pandas.pydata.org/docs/reference/api/pandas.read_parquet.html works fine.

Let's use this dataset then so that we have something that looks a little more realistic than the metrics json:

```python
>>> from datasets import load_dataset
>>> load_dataset(
...     "csv",
...     data_files="dvc://workshop/satellite-data/jan_train.csv",
...     storage_options={"url": "https://github.com/iterative/dataset-registry.git"}
... )
Downloading data: 100%|███████████████████████████████| 132M/132M [00:30<00:00, 4.32MB/s]
Downloading data files: 100%|██████████████████████████████| 1/1 [00:32<00:00, 32.17s/it]
Extracting data files: 100%|██████████████████████████████| 1/1 [00:00<00:00, 443.84it/s]
Generating train split: 503227 examples [00:00, 514266.75 examples/s]
DatasetDict({
    train: Dataset({
        features: ['id', 'epoch', 'sat_id', 'x', 'y', 'z', 'Vx', 'Vy', 'Vz', 'x_sim', 'y_sim', 'z_sim', 'Vx_sim', 'Vy_sim', 'Vz_sim'],
        num_rows: 503227
    })
})
```